### ISSUE TITLE
Error in ELHelper? (trimou-extension-el 2.0.3)

### ISSUE THREAD

**Issue by rmuller** [trimou/trimou#113]
Template:
````
{{#el "[1,2,3]"}}
Number {{this}}
{{/el}}
````
Output:
````
Number [1,2,3]
````
Expected output:
````
Number 1
Number 2
Number 3
````
The output created by trimou is the output i expect for template 
`Number {{el "[1,2,3]"}}`

**Comment by mkouba** [comment]
I don't think it's a bug. `ELHelper` does not iterate over the evaluated result - it only pushes the result on the stack (see also javadoc). If you want to iterate then:
````
{{#el "[1,2,3]"}}
  {{#each}}
    Number {{this}}
  {{/each}}
{{/el}}
````
Maybe we could introduce `ELEachHelper` and override `EachHelper` similarly as `ELIfHelper` overrides the regular `IfHelper`.

**Comment by rmuller** [comment]
Thanks again for the fast reply.

I checked the Javadoc and indeed this is not an error. My bad. However, is this consistent with "normal" mustache behavior? `{{#expression}}`  behaves different if `expression` resolves to a Collection or not.

**Comment by mkouba** [comment]
> However, is this consistent with "normal" mustache behavior? 

Well, here we're talking about a helper (Handlebars), not a regular section tag (Mustache). I think `ELEachHelper` might be a good idea though.

**Comment by rmuller** [comment]
You are right. And yes, `ELEachHelper` is in fact "the missing link".

**Comment by rmuller** [comment]
Just to be sure, so this is the best (efficient /elegant) way to do it? Or does trimou offer a better approach?

```
  {{#el "expensiveExpression"}} {{! returns list of elements, possible empty }}
  {{#if this}}
  <OUTER_ELEMENT> {{! should only be outputted if at least one element is available }}
    {{#each}}
    <INNER_ELEMENT>
      {{! other logic }}
    </INNER_ELEMENT>
    {{/each}}
  </OUTER_ELEMENT>
  {{/if}}
  {{/el}}
```

**Comment by mkouba** [comment]
Hm, we should probably also fix `EachHelper` so that it handles `null` object from the context stack as an empty object. And then `{{#if this}}` would be unnecessary.

**Comment by mkouba** [comment]
Fixed by https://github.com/trimou/trimou/commit/60b348ad51714a4d2ae258240b2c06ba757351a2

**Comment by mkouba** [comment]
Ah, sorry I overlooked the `<OUTER_ELEMENT>` part. In this case, you cannot skip the `{{#if}}` part.

**Comment by rmuller** [comment]
Thanks for confirming. Nevertheless, fix 60b348a is a good one.
